### ISSUE TITLE
Amplify cli tries to create Notifications even though they already exist.

### ISSUE THREAD

**Issue by ericstoecker** [aws-amplify/amplify-cli#11270]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v14.17.0

### Amplify CLI Version

10.3.2

### What operating system are you using?

macOS Monterey 12.6

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No manual changes made

### Amplify Categories

analytics, notifications

### Amplify Commands

pull, push

### Describe the bug

- Setup push notifications with the amplify cli (APNS and FCM) and push to the cloud.
- When pulling this on a different device (also using amplify cli) amplify wants to "create" the notifications category again.
However this happened multiple months after the notifications were created, so multiple pull and push operations worked without causing any problems.
- When pulling or pushing amplify asks me to choose an authentication method for APNs. I don't want to do this because everything should already be set up in the cloud.

### Expected behavior

after adding and setting up notifications on one device and pushing to the cloud, amplify should not try to set notifications up on a different device.

### Reproduction steps

1. create project and set up notifications with APNs and FCM on a device and push to cloud.
2. on another device set up amplify cli and pull everything.
3. cli will ask to choose authentication method (I then aborted the command)
<img width="522" alt="image" src="https://user-images.githubusercontent.com/71395664/198235594-030b524f-8a35-4ddb-b496-8277b9bbdd59.png">
4. my amplify-meta.json now contains the following 
<img width="229" alt="image" src="https://user-images.githubusercontent.com/71395664/198232088-cbfa8567-a63b-4cbb-87de-8e1ded6693ac.png">

5. as you can see there is no "lastPushTimeStamp" so amplify probably believes that this resource is newly created. Also there is no other information except for the channels and service
6. run `amplify status`
7. should now look something like this
<img width="677" alt="image" src="https://user-images.githubusercontent.com/71395664/198232840-0991500b-fde7-4109-afa7-5cafdd89a551.png">
8. run `amplify pull`
9. cli will ask to choose authentication method (I then aborted the command)
<img width="522" alt="image" src="https://user-images.githubusercontent.com/71395664/198235594-030b524f-8a35-4ddb-b496-8277b9bbdd59.png">

10. now the amplify-meta.json contains the following
<img width="590" alt="image" src="https://user-images.githubusercontent.com/71395664/198234731-129f89ca-abfb-4a6d-8d15-762dcee13ec7.png">

11. setting "HasTokenKey" to "true" leads to amplify not trying to create notifications category 
<img width="665" alt="image" src="https://user-images.githubusercontent.com/71395664/198236494-6ece6467-d1eb-4ed3-8569-7ac31825571a.png">

12. now change something in your project to be able to run `amplify push` (for example you can simply add a space in your schema.graphql)
13. when running `amplify pull` again the cli again asks me again to choose an authentication method for APNs
14. so theoretically I can push without recreating notifications. However this is not the expected behaviour and I don't want to do the above steps each time I have to push something.
15. I have also figured out that for some reason running `amplify env pull` instead of just `amplify pull` leads to my amplify-meta.json looking again like this
<img width="345" alt="image" src="https://user-images.githubusercontent.com/71395664/198241433-06f78ab8-2a65-4837-a05e-1040558be9c3.png">

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Project Identifier

Project Identifier: 8e7ec6dac606d8af0b67897bb966c442

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Additional information

Seems to be related to (or exactly the same as) #6408 but I couldn't comment or reopen this issue since it was closed.
The solution there was to recreate the whole project. Since my project is already in production I would rather not do that.

**Comment by Mentioum** [comment]
We're also having this issue while using `v10.x.x`: 

It resulted in us having too many Pinpoint policies being added to our auth role.  
This prevented us pushing after a while with the error: `can't have more than 10 policies per role`

We manually deleted the invalid policies and duplicated pinpoint projects and downgraded to `v9.2.1` until this is fixed.

Tagging @ralphkemp for reference.

**Comment by Straubulous** [comment]
@ericstoecker  @Mentioum  - we're investigating

**Comment by lazpavel** [comment]
Hi @ericstoecker @Mentioum we have a RC out with a fix for this issue which you can find here: https://www.npmjs.com/package/@aws-amplify/cli/v/10.4.0-rc.43121b9b9.0 and we are working to push it to latest

**Comment by josefaidt** [comment]
Hey @ericstoecker it appears this was fixed with CLI [10.4.0](https://github.com/aws-amplify/amplify-cli/releases/tag/v10.4.0), are you still experiencing this issue?

**Comment by ericstoecker** [comment]
> Hey @ericstoecker it appears this was fixed with CLI [10.4.0](https://github.com/aws-amplify/amplify-cli/releases/tag/v10.4.0), are you still experiencing this issue?

fixed it for me too

**Comment by RalphKemp** [comment]
Thanks for fixing this so quickly, I'll give it a test today. Cheers
